Commonalities/documentation/Working/user-consent.md

```diff
@@ -0,0 +1,405 @@
+
+![Purpose association to App](./images/purpose-association-to-app.png)
+
+Therefore, an application will have access to:
+  - APIs or resources covered by the assigned scopes
+  - APIs or resources providing personal information under the assigned purpose (because of the PI scopes included in that purpose)
+    - **Only** if user has granted her consent for that purpose
+    - Which may happen automatically (automatic purposes) or may need an explicit user action to provide consent (depending on the purpose level of the purpose associated to that consent). In the second case, the consent is created and stored and will be validated during authorization.
+
+## Using purpose parameter in the authorization request
+
+The application can request a list of purposes in the authorization request of the [OAuth 2.0 Authorization Framework](https://www.rfc-editor.org/rfc/rfc6749) using an additional `purpose` parameter:
+
+| **Authorize request param** | Description |
+| --- | --- |
+| `purpose` | The value of this parameter is expressed as a list of space-delimited, case-sensitive strings. The strings will be the id of the purposes. |
+
+This parameter has been defined in these guidelines to allow applications to provide a purpose/s name which represents the reason why a client application needs to process a certain piece of user Personal Information. Note that, as explained before, a purpose maps to a list of scopes giving access to Personal Information resources.
+
+_Example: requesting "purpose\_1", "purpose\_2" and "purpose\_3"_
+
+```
+POST /bc-authorize HTTP/1.1
+Authorization: Basic {Credentials}
+Content-Type: application/x-www-form-urlencoded
+
+purpose=purpose_1%20purpose_2%20purpose_3&
+login_hint_token=eyJ[…].efg[…].asW[…]&
+… 
+```
+
+The access token obtained from Token Endpoint will be granted to a subset or to all the scopes included in the requested purposes. The authorization server MAY fully or partially ignore the scopes of the requested purposes based on the authorization server policy. The authorization server MUST include the obtained scope list in the `scope` response parameter of the Token Endpoint to inform the client of the actual scopes granted.
+
+The authorization server MUST include the obtained purpose list in the `purpose` response parameter of the Token Endpoint. The purpose MUST only be included in the response if at least one scope of the purpose has been granted.
+
+_Example: only scopes of the "purpose\_1" and "purpose\_2" have been granted_
+
+```
+POST /token HTTP/1.1
+Authorization: Basic {Credentials}
+Content-Type: application/x-www-form-urlencoded
+
+grant_type=urn%3Aopenid%3Aparams%3Agrant-type%3Aciba&
+auth_req_id=1c266114-a1be-4252-8ad1-04986c5b9ac1
+
+HTTP/1.1 200 OK
+Content-Type: application/json;charset=UTF-8
+
+{
+  "access_token":"2YotnFZFEjr1zCsicMWpAA",
+  "token_type":" Bearer",
+  "expires_in":3600,
+  "scope":"scope_1_of_purpose_1 scope_2_of_purpose_1 scope_1_of_purpose_2",
```

Comment:
Same comment as above on space and comma.

Reply:
It follows OAuth 2.0 standard. https://www.rfc-editor.org/rfc/rfc6749
```The value of the scope parameter is expressed as a list of space-delimited, case-sensitive strings.```